File: POO/TiposClases/src/Animals/Duck.java
package Animals;

public class Duck extends Animal implements Flyer{
    String name;

    public Duck(String name) {
        this.name = name;
    }

    @Override
    public void makeSound() {
        System.out.println(name + " says cuak...");
    }

    @Override
    public void fly() {
        System.out.println(name + " is flying...");
    }
    
}

File: 1.SRP/src/main/kotlin/com/example/animal/Duck.kt
package com.example.animal

import java.lang.reflect.Type

class Duck (
    type: String,
    name: String,
    age: Int?,
    size: String
) : Animal(type, name, age), Volador {

    override fun fly() {
        println("The duck $name is flying...")
    }

    override fun eat() {
        println("The duck $name is eating")
    }

    override fun talking() {
        println("Duck $name says cuac...")
    }
}